### DISCUSSION TITLE
Github pull_request doesn't triggered on commits created by other actions

### DISCUSSION THREAD

**Discussion by huydx** [community/community#63623] [Actions]
### Select Topic Area

Bug

### Body

Hello there, not sure it's a bug so just trying to explain what happen on our side.
We have a actions which does create pull request programmatically  using `actions/github-script` (by calling `github.rest.pulls.create`).
What weird here is: The pull request doesn't trigger our test action (which based on `on: pull_request` event) for its last commit.

Some detail on our setup
- create-pull-request.yml:  use `actions/github-script`  to create pull request programatically (weekly)
- unit-test.yml which trigger on `on: pull_request` event

Expected behavior:
- when `create-pull-request.yml` ran it should trigger unit-test.yml

Actual behavior:
- `create-pull-request.yml` wasn't trigger unit-test.yml

Any way / hint for me to debug this situation?

Many thanks!

**Comment by Abdullahxz** [comment]
@huydx When you use the repository's `GITHUB_TOKEN` to perform tasks, events triggered by the `GITHUB_TOKEN`, with the exception of `workflow_dispatch` and `repository_dispatch`, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's `GITHUB_TOKEN`, a new workflow will not run even when the repository contains a workflow configured to run when `push` events occur.

If you do want to trigger a workflow from within a workflow run, you can use a GitHub App installation access token or a personal access token instead of `GITHUB_TOKEN` to trigger events that require a token.

You can read more about this in [documentation](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow).
